lib/std/Build/Step/Compile.zig

```diff
@@ -1140,17 +1161,28 @@ fn make(step: *Step, prog_node: *std.Progress.Node) !void {
                             for (c_source_file.flags) |arg| {
                                 try zig_args.append(arg);
                             }
+                            if (c_source_file.precompiled_header) |pch| {
+                                try zig_args.append("-include-pch");
+                                try zig_args.append(pch.getEmittedBin().getPath(b));
+                                try zig_args.append("-fpch-validate-input-files-content");
+                            }
                             try zig_args.append("--");
                             prev_has_cflags = true;
                         }
+
+                        if (self.kind == .pch) {
+                            try zig_args.append("-x");
+                            try zig_args.append(if (self.is_linking_libcpp) "c++-header" else "c-header");
```

Comment:
is this controlling the input language or the output language? In either case, whether libc++ is being linked is not the correct logic.

Shouldn't this depend on the file extension of the input header instead?